db/compaction/compaction_picker.cc

```diff
@@ -1126,7 +1126,11 @@ void CompactionPicker::RegisterCompaction(Compaction* c) {
          c->output_level() == 0 ||
          !FilesRangeOverlapWithCompaction(*c->inputs(), c->output_level(),
                                           c->GetPenultimateLevel()));
-  if (c->start_level() == 0 ||
+  // File ingestion's start level is just a placeholder number without actual
+  // meaning as file ingestion technically does not have an input level like
+  // other compactions
+  if ((c->start_level() == 0 &&
+       c->compaction_reason() != CompactionReason::kExternalSstIngestion) ||
```

Comment:
File ingestion is treated as a Compaction with input level L0 as a placeholder number because technically file ingestion does not have an input level and we need a number for populating Compaction.

(I thought of using -1 but its actually the Compaction::kInvalidLevel used for error checking or exclusion. )

As a result of this placeholder number "0", we can't see file ingestion as level 0 compaction otherwise every file ingestion can block other innocent compactions (e.g, https://github.com/facebook/rocksdb/blob/7.8.fb/db/compaction/compaction_picker.cc#L646, https://github.com/facebook/rocksdb/blob/7.8.fb/db/compaction/compaction_picker_level.cc#L716.